File: APIRollTheDice-Backend/src/main/java/com/example/APIRollTheDice/APIRollTheDice_Backend/Model/DTO/Chat/ChatMessageDTO.java
package com.example.APIRollTheDice.APIRollTheDice_Backend.Model.DTO.Chat;


import java.time.LocalDateTime;

public class ChatMessageDTO {
    private Long id;

    private Long idSender;

    private String messageContent;

    private boolean isModified=false;

    private LocalDateTime sentAt = LocalDateTime.now();

    private Long idChatChanel;



    public ChatMessageDTO() {
    }

    public ChatMessageDTO(Long id, Long idSender, String messageContent, boolean isModified, LocalDateTime sentAt, Long idChatChanel) {
        this.id = id;
        this.idSender = idSender;
        this.messageContent = messageContent;
        this.isModified = isModified;
        this.sentAt = sentAt;
        this.idChatChanel = idChatChanel;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getIdSender() {
        return idSender;
    }

    public void setIdSender(Long idSender) {
        this.idSender = idSender;
    }

    public String getMessageContent() {
        return messageContent;
    }

    public void setMessageContent(String messageContent) {
        this.messageContent = messageContent;
    }

    public boolean isModified() {
        return isModified;
    }

    public void setModified(boolean modified) {
        isModified = modified;
    }

    public LocalDateTime getSentAt() {
        return sentAt;
    }

    public void setSentAt(LocalDateTime sentAt) {
        this.sentAt = sentAt;
    }

    public Long getIdChatChanel() {
        return idChatChanel;
    }

    public void setIdChatChanel(Long idChatChanel) {
        this.idChatChanel = idChatChanel;
    }
}

File: RollTheDice/Assets/_Project/API/Model/DTO/ChatDTO/ChatMessageDTO.cs
using Assets._Project.API.Model.Object.User;
using System;

namespace Assets._Project.API.Model.DTO.ChatDTO
{
    public class ChatMessageDTO
    {
        public long Id { get; set; }
        public UserIdentifantData IdSender { get; set; }

        public string MessageContent { get; set; }

        public bool IsModified { get; set; }

        public DateTime SentAt { get; set; }
        public long IdChatChanel { get; set; }

        public ChatMessageDTO(){}
        public ChatMessageDTO(long id, UserIdentifantData idSender, long idChatChanel, DateTime dateTime, bool isModified)
        {
            Id = id;
            IdSender = idSender;
            IdChatChanel = idChatChanel;
            SentAt = dateTime;
            IdChatChanel = idChatChanel;
            IsModified = isModified;
        
        }
    }
}